contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -4658,9 +4734,17 @@ static void post_process_table_source(TSqlParser::Table_source_itemContext *ctx,
 	for (auto cctx : ctx->table_source_item())
 		post_process_table_source(cctx, expr, baseCtx);
 
-	for (auto wctx : ctx->with_table_hints())
+	for (auto wctx : ctx->with_table_hints()) {
+		if(!wctx->sample_clause()) {
+			std::string table_name;
+			if(ctx->full_object_name())
+				table_name = ::getFullText(ctx->full_object_name());
+			else
```

Comment:
There are cases other than `full_object_name` or `local_id`. To prevent unexpected results, let's call `extractTableHints()` only if we have a valid table_name.